setup-unattended.ps1

```diff
@@ -0,0 +1,133 @@
+#
+# This script will install or up
+#
+#
+[CmdletBinding()]
+param (
+    [Parameter(HelpMessage = "The enclave enrolment key to use when enrolling")]
+    [string]
+    $EnrolmentKey
+)
+
+# Hide progress bars to speed up downloads
+$ProgressPreference = 'SilentlyContinue';
+$ErrorActionPreference = 'Stop';
+
+$manifestDoc = Invoke-WebRequest -Uri "https://install.enclave.io/manifest/windows/setup-unattended-msi.json" | ConvertFrom-Json;
+
+$latestGaRelease = $manifestDoc.ReleaseVersions | Where-Object ReleaseType -EQ "GA" | Select-Object -Last 1;
+
+$newEnclaveVersion = "$($latestGaRelease.MajorVersion).$($latestGaRelease.MinorVersion).$($latestGaRelease.BuildVersion).$($latestGaRelease.RevisionVersion)";
+
+$ErrorActionPreference = 'SilentlyContinue';
+
+$enclaveBinaryPath = "C:\Program Files\Enclave Networks\Enclave\Agent\bin\enclave.exe";
+$enclaveTrayPath = "C:\Program Files\Enclave Networks\Enclave\Agent\bin\enclave-tray.exe";
+
+$existingEnclaveVersion = $(get-command $enclaveBinaryPath);
+
+$existingEnclaveVersion = ($existingEnclaveVersion.Version).ToString();
+
+$ErrorActionPreference = 'Stop';
+
+if ($newEnclaveVersion -ne $existingEnclaveVersion)
+{
+    # We need to install.
+    # Get the architecture
+    $systemArch = [System.Environment]::GetEnvironmentVariable('PROCESSOR_ARCHITECTURE', 'Machine');
+    $enclaveArch = '';
+    $vcRedistUrl = '';
+
+    if ($systemArch -eq 'AMD64')
+    {
+        $enclaveArch = 'X64';
+        $vcRedistUrl = 'https://aka.ms/vs/17/release/vc_redist.x64.exe';
+    }
+    elseif ($systemArch -eq 'ARM64')
+    {
+        $enclaveArch = 'Arm64';
+        $vcRedistUrl = 'https://aka.ms/vs/17/release/vc_redist.arm64.exe';
+    }
+    elseif ($systemArch -eq 'X86')
+    {
+        $enclaveArch = 'X86';
+        $vcRedistUrl = 'https://aka.ms/vs/17/release/vc_redist.x86.exe';
+    }
+
+    if ($enclaveArch -eq '')
+    {
+        Write-Error "Unsupported architecture $systemArch";
+        return 1;
+    }
+
+    # Now grab the latest installer.        
+    $selectedPackage = $latestGaRelease.Packages | Where Architecture -EQ $enclaveArch;
+
+    if (!$existingEnclaveVersion)
+    {
+        "Installing VC++ Redistributable if needed";
+
+        $vcRedistInstallPath = Join-Path $env:TEMP "enclave_vcredist.exe";
+    
+        # No existing version; install latest VC++ (will quickly complete if it already exists)
+        Invoke-WebRequest -Uri $vcRedistUrl -OutFile $vcRedistInstallPath;
+
+        & $vcRedistInstallPath "/install" "/silent" "/norestart";
+    }
+
+    "Downloading latest enclave version from $($selectedPackage.Url)";
+
+    $enclaveInstallerFile = Join-Path $env:TEMP "enclave-$newEnclaveVersion.msi";
+    $enclaveInstallLogFile = Join-Path $env:TEMP "enclave-$newEnclaveVersion.install.log"
+
+    Invoke-WebRequest $selectedPackage.Url -OutFile $enclaveInstallerFile;
+
+    if ($existingEnclaveVersion)
+    {
+        "Closing any open instances of the enclave tray"
```

Comment:
why are we just closing the tray should we not stop encalve entirely?

Reply:
The installer does that for us, but it doesn't close the tray.